test/e2e/clusterctl_upgrade.go

```diff
@@ -334,6 +336,18 @@ func ClusterctlUpgradeSpec(ctx context.Context, inputGetter func() ClusterctlUpg
 
 		By("THE MANAGEMENT CLUSTER WITH THE OLDER VERSION OF PROVIDERS IS UP&RUNNING!")
 
+		machineCRD := &apiextensionsv1.CustomResourceDefinition{}
+		if err := managementClusterProxy.GetClient().Get(ctx, client.ObjectKey{Name: "machines.cluster.x-k8s.io"}, machineCRD); err != nil {
+			fmt.Printf("Failed to get machine CRD %s: %v\n", machineCRD.Name, err)
```

Comment:
Let's please fail here instead of just printing the error

Hm also machineCRD.Name will be probably empty if we hit an error

Reply:
(if we hit the error case we probably panic in l.347 atm)